src/fastfield/serializer/gcd.rs

```diff
@@ -0,0 +1,176 @@
+use fastdivide::DividerU64;
+use gcd::Gcd;
+
+pub const GCD_DEFAULT: u64 = 1;
+
+pub fn find_gcd(mut numbers: impl Iterator<Item = u64>, base: u64) -> Option<u64> {
+    let mut num1 = 0;
+    let mut num2 = 0;
+    loop {
+        let num = numbers.next()? - base;
+        if num1 == 0 {
+            num1 = num;
+        }
+        if num2 == 0 {
+            num2 = num;
+        }
+        if num1 != 0 && num2 != 0 {
+            break;
+        }
+    }
+    let mut gcd = (num1).gcd(num2);
```

Comment:
I'm not sure what the gcd crate returns on value 0, but we want `gcd(0, n) = n`

Reply:
with the 0-filter this case is covered